### PR TITLE
fix: Some small fixes.

### PR DESCRIPTION
**Pull request by feanil**
* Be able to handle brand new repos that have no branches in them.
* Change the order we look through all the repos so we check the newest
  cretaed repos first.

### CODE REVIEW

File: migrate/repo_checks.py [hunk 1]
```diff
@@ -494,7 +494,16 @@ def _update_branch_protection(self, params):
...
 
-        resp.raise_for_status()
+        try:
+            resp.raise_for_status()
+        except requests.HTTPError as e:
+            # If the branch is missing, the repo is probably new and has no branches
+            # ignore that for now.
+            if not (
+                "message" in resp.json()
+                and resp.json()["message"] == "Branch not found"
```
Comment (kdmccormick) [inline migrate/repo_checks.py:504]:
```suggestion
            if resp.json.get("message") != "Branch not found":
```

**Review by kdmccormick — COMMENTED** [review]
Looks good. Will this handle repos with a default branch other than master, eg main or release? I wouldn't want the check to pass silently because someone got creative with their default branch name, causing master/main to not be found.

**Comment by feanil** [comment]
@kdmccormick yes, it will, we ask github for the default branch and we try to use it here, so the cases when this happens is when the named default branch does not exist in the repo yet.

**Review by kdmccormick — APPROVED** [review]
thanks!

File: migrate/repo_checks.py [hunk 2]
```diff
@@ -494,7 +494,14 @@ def _update_branch_protection(self, params):
...
         resp = requests.put(url, headers=headers, json=params)
 
-        resp.raise_for_status()
+        try:
+            resp.raise_for_status()
+        except requests.HTTPError as e:
+            # If the branch is missing, the repo is probably new and has no branches
+            # ignore that for now.
+            if resp.json.get("message") != "Branch not found":
+            ):
```
Comment (regisb) [inline migrate/repo_checks.py:503]: This is a syntax error. I can open a PR to fix it, but I'm wondering: when and where is this script run? If it did run, did it trigger an error?

Comment (feanil) [reply migrate/repo_checks.py:503]: @regisb thanks for the catch, the script is still being run manually but in the future will hopefully run as a github action periodically.  I can take care of fixing before we run it again.

Comment (feanil) [reply migrate/repo_checks.py:503]: https://github.com/openedx/terraform-github/pull/55

**Pull request merged**